packages/block-editor/src/components/use-settings/index.js

```diff
@@ -0,0 +1,272 @@
+				if ( blockedPaths.includes( path ) ) {
+					// eslint-disable-next-line no-console
+					console.warn(
+						'Top level useSetting paths are disabled. Please use a subpath to query the information needed.'
+					);
+					return undefined;
+				}
+
+				// 0. Allow third parties to filter the block's settings at runtime.
+				let result = applyFilters(
+					'blockEditor.useSetting.before',
+					undefined,
+					path,
+					clientId,
+					blockName
+				);
+
+				if ( undefined !== result ) {
+					return result;
+				}
+
+				const normalizedPath = removeCustomPrefixes( path );
+
+				// 1. Take settings from the block instance or its ancestors.
+				// Start from the current block and work our way up the ancestors.
+				for ( const candidateClientId of candidates ) {
+					const candidateAtts =
+						select( blockEditorStore ).getBlockAttributes(
+							candidateClientId
+						);
+					result =
+						getValueFromObjectPath(
+							candidateAtts.settings?.blocks?.[ blockName ],
+							normalizedPath
+						) ??
+						getValueFromObjectPath(
+							candidateAtts.settings,
+							normalizedPath
+						);
+					if ( result !== undefined ) {
+						// Stop the search for more distant ancestors and move on.
+						break;
+					}
+				}
+
+				// 2. Fall back to the settings from the block editor store (__experimentalFeatures).
+				const settings = select( blockEditorStore ).getSettings();
+				if ( result === undefined && blockName ) {
+					result = getValueFromObjectPath(
+						settings.__experimentalFeatures?.blocks?.[ blockName ],
+						normalizedPath
+					);
+				}
+
+				if ( result === undefined ) {
+					result = getValueFromObjectPath(
+						settings.__experimentalFeatures,
+						normalizedPath
+					);
+				}
+
+				// Return if the setting was found in either the block instance or the store.
+				if ( result !== undefined ) {
+					if ( PATHS_WITH_MERGE[ normalizedPath ] ) {
+						return mergeOrigins( result );
+					}
+					return result;
+				}
+
+				// 3. Otherwise, use deprecated settings.
+				const deprecatedSettingsValue =
+					deprecatedFlags[ normalizedPath ]?.( settings );
+				if ( deprecatedSettingsValue !== undefined ) {
+					return deprecatedSettingsValue;
+				}
+
+				// 4. Fallback for typography.dropCap:
+				// This is only necessary to support typography.dropCap.
+				// when __experimentalFeatures are not present (core without plugin).
+				// To remove when __experimentalFeatures are ported to core.
+				return normalizedPath === 'typography.dropCap'
+					? true
+					: undefined;
+			} );
+		},
+		[ blockName, clientId, paths ]
+	);
+}
+
+/**
+ * Hook that retrieves the given setting for the block instance in use.
+ *
+ * It looks up the setting first in the block instance hierarchy.
+ * If none is found, it'll look it up in the block editor settings.
+ *
+ * @param {string} path The path to the setting.
+ * @return {any} Returns the value defined for the setting.
+ * @deprecated
```

Comment:
Should we add the version here, like in other places where we use the `@deprecated` tag?

Reply:
Version and note added :+1: